packages/framework/src/DependencyInjection/Compiler/RegisterProjectFrameworkClassExtensionsCompilerPass.php

```diff
@@ -44,4 +63,29 @@ private function isProjectClass(string $serviceId): bool
     {
         return strpos($serviceId, 'App') !== false;
     }
+
+    /**
+     * @param \Roave\BetterReflection\Reflection\ReflectionClass $frameworkClassBetterReflection
+     * @param \Symfony\Component\DependencyInjection\Definition $classExtensionRegistryDefinition
+     * @param string $aliasId
+     */
+    private function addAllVariantsOfServiceToClassExtension(
+        ReflectionClass $frameworkClassBetterReflection,
+        Definition $classExtensionRegistryDefinition,
+        string $aliasId
+    ): void {
+        if ($frameworkClassBetterReflection->isInterface()) {
+            $className = rtrim($frameworkClassBetterReflection->getName(), 'Interface');
+
+            if (class_exists($className)) {
+                $classExtensionRegistryDefinition->addMethodCall('addExtendedService', [$className, $aliasId]);
+            }
+        } else {
+            $interfaceName = $frameworkClassBetterReflection->getName() . 'Interface';
+
+            if (interface_exists($interfaceName)) {
```

Comment:
Wouldn't be nice also to check whether the class implements the interface?

If I just create the `Shopsys\FrameworkBundle\Model\Cart\Watcher\CartWatcherFacadeInterface` interface, the link between this interface and ordinary CartWatcherFacade will be created in the service extension map, even though it does not make sense at this point

Reply:
Added